src/it/scala/org/apache/spark/sql/snowflake/SnowflakeSqlImplicits.scala

```diff
@@ -0,0 +1,243 @@
+package org.apache.spark.sql.snowflake
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
+import org.apache.spark.sql.{ColumnName, Encoder, Encoders}
+
+import scala.collection.Map
+import scala.language.implicitConversions
+import scala.reflect.runtime.universe.TypeTag
+
+/**
+ * A collection of implicit methods for converting common Scala objects into [[org.apache.spark.sql.Dataset]]s.
+ *
+ * @since 1.6.0
+ */
+abstract class SnowflakeSQLImplicits(sfSession: SFTestWrapperSparkSession)
```

Comment:
Why do we need to copy these over ? If we are extending session, don't the implicits just apply ?

Reply:
Because of the way implicits work in Scala, if I inherited the superclass test suite, the implicits defined *there* would apply, instead of anything I tried to override. So instead, I need to copy/paste all the test code from the suite that this copies.

I also couldn't just subclass `SqlImplicits`, because the method would override the `Seq -> DatasetHolder` implicit conversion defined there, and `DatasetHolder` is a case class, which couldn't be subclassed: meaning that the `Seq` had to be converted into `DatasetHolder`, and `DatasetHolder` is effectively final.

That's how I settled on copy/pasting the implicits definitions instead of using any inheritance.